src/bundles/files/actions.js

```diff
@@ -398,6 +398,37 @@ const actions = () => ({
     }
   }),
 
+  /**
+   * Adds CAR file. On completion will trigger `doFilesFetch` to update the state.
+   * @param {FileStream} carFile
+   * @param {string} name
+   */
+  doAddCarFile: (carFile, name = '') => perform(ACTIONS.ADD_CAR_FILE, async (ipfs, { store }) => {
+    ensureMFS(store)
+
+    const fileResult = await fileToByteArray(carFile)
+    try {
+      const result = await all(ipfs.dag.import([fileResult], {
```

Comment:
You should be able to pass a stream of the file to dag.import in order to avoid loading the whole CAR into memory.